FAILED: [code=1] Utilities/cmcurl/curltest 
: && .../host/bin/gcc -O2 -I.../host/include -w -O3 -DNDEBUG -L.../host/lib Utilities/cmcurl/CMakeFiles/curltest.dir/curltest.c.o -o Utilities/cmcurl/curltest  Utilities/cmcurl/lib/libcmcurl.a  .../host/lib/libssl.a  .../host/lib/libcrypto.a  -lpthread  .../host/lib/libz.a  Utilities/cmnghttp2/libcmnghttp2.a && :
/bin/ld: .../host/lib/libcrypto.a(libcrypto-lib-dso_dlfcn.o): in function `dlfcn_globallookup':
dso_dlfcn.c:(.text+0x13): undefined reference to `dlopen'
/bin/ld: dso_dlfcn.c:(.text+0x26): undefined reference to `dlsym'
/bin/ld: dso_dlfcn.c:(.text+0x31): undefined reference to `dlclose'
/bin/ld: .../host/lib/libcrypto.a(libcrypto-lib-dso_dlfcn.o): in function `dlfcn_bind_func':
dso_dlfcn.c:(.text+0x1e7): undefined reference to `dlsym'
/bin/ld: dso_dlfcn.c:(.text+0x2de): undefined reference to `dlerror'
/bin/ld: .../host/lib/libcrypto.a(libcrypto-lib-dso_dlfcn.o): in function `dlfcn_load':
dso_dlfcn.c:(.text+0x351): undefined reference to `dlopen'
/bin/ld: dso_dlfcn.c:(.text+0x3d8): undefined reference to `dlclose'
/bin/ld: dso_dlfcn.c:(.text+0x411): undefined reference to `dlerror'
/bin/ld: .../host/lib/libcrypto.a(libcrypto-lib-dso_dlfcn.o): in function `dlfcn_pathbyaddr':
dso_dlfcn.c:(.text+0x4d2): undefined reference to `dladdr'
/bin/ld: dso_dlfcn.c:(.text+0x543): undefined reference to `dlerror'
/bin/ld: .../host/lib/libcrypto.a(libcrypto-lib-dso_dlfcn.o): in function `dlfcn_unload':
dso_dlfcn.c:(.text+0x6d4): undefined reference to `dlclose'
collect2: error: ld returned 1 exit status

---


--- a/Utilities/cmcurl/CMakeLists.txt
+++ b/Utilities/cmcurl/CMakeLists.txt
@@ -1187,6 +1187,12 @@ if(CURL_DEFAULT_SSL_BACKEND AND NOT _val
   message(FATAL_ERROR "CURL_DEFAULT_SSL_BACKEND '${CURL_DEFAULT_SSL_BACKEND}' not enabled.")
 endif()
 
+# Add libdl
+check_library_exists("dl" dlopen "" HAVE_LIBDL)
+if(HAVE_LIBDL)
+  list(APPEND CURL_LIBS "dl")
+endif ()
+
 # Keep ZLIB detection after TLS detection,
 # and before calling curl_openssl_check_exists().
 
